.github/workflows/bun-support.yml

```
@@ -0,0 +1,83 @@
+name: Bun Support
+#on:
+#  schedule:
+#    - cron: 0 0 * * *
+#  workflow_dispatch:
+
+on:
+  push:
+    branches: [master, develop]
+    tags: ['*']
+  pull_request:
+    types: [opened, reopened, synchronize]
+  workflow_dispatch:
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - uses: oven-sh/setup-bun@v1
+
+      - run: bunx ci --omit=peer
+
+      - name: Test Block
+        run: bunx vitest run test
+        working-directory: packages/block
+
+      - name: Test Blockchain
+        run: bunx vitest run test
+        working-directory: packages/blockchain
+
+      # Client not working yet in Bun (expected), 2024-01-08
+      # - name: Test Client
+      #   run: bunx vitest run test
+      #   working-directory: packages/client
+
+      - name: Test Common
+        run: bunx vitest run test
```

I don't think this is actually using bun. Because vitest is an executable bun should default to using node here. I think you need to pass in the `--bun` flag to use bun and you might have issues related more to vitest than bun.
```suggestion
        run: bun --bun run vitest run test
```

---

Docs on this are here https://bun.sh/docs/cli/run#bun

---

Oh, that is the "tip of the year"! 🙏 🙂 

I was already wondering why things are so smooth. 😂 That is the missing link I needed!

---

Just to note that this is not yet working (using bun 1.0.26). Tested with `common` (counter test with the original version worked), new version just hangs before any output.

Not dropping here for you to debug, might just be our homework to do with the bun updates adjusting our code base. But just to drop the information here.